bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -635,7 +635,9 @@ func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, config
 	}
 
 	if err := r.Client.Create(ctx, secret); err != nil {
-		return errors.Wrapf(err, "failed to create kubeconfig secret for KubeadmConfig %s/%s", config.Namespace, config.Name)
+		if !apierrors.IsAlreadyExists(err) {
```

Comment:
In the event of the boostrapConfig getting updated, by a cluster operator, and the changes the bootstrapConfig's status. It seems like we could end up with a mismatch between boostrapConfig's `Status` and the secret's `Data` and this would leave the bootstrap config data to be inconsistent. I think that the right fix would be to update the secret if it exists.

Reply:
It is not valid for anyone or anything outside of the kubeadm bootstrapper to generate or manipulate the bootstrap data secret. This is unsupported.

Reply:
Thanks for clarifying! I was thinking if it would be useful to detect that and let the user know.